hrm-domain/hrm-core/profile/profileRoutesV0.ts

```diff
@@ -40,99 +40,71 @@ const profilesRouter = SafeRouter();
  * @param {profileController.ProfileListConfiguration['offset']} req.query.offset - Offset
  * @param {profileController.SearchParameters['filters']['profileFlagIds']} req.query.profileFlagIds
  */
-profilesRouter.get(
-  '/',
-  publicEndpoint,
-  async (req: Request, res: Response, next: NextFunction) => {
-    try {
-      const { accountSid } = req;
-      const {
-        sortDirection,
-        sortBy,
-        limit,
-        offset,
-        profileFlagIds: encodedProfileFlagIds,
-      } = req.query as Record<string, any>; // TODO: maybe improve this validation
-
-      const profileFlagIds = encodedProfileFlagIds
-        ? decodeURIComponent(encodedProfileFlagIds)
-            .split(',')
-            .map(s => parseInt(s, 10))
-            .filter(v => v && !isNaN(v))
-        : undefined;
-
-      const filters = {
-        profileFlagIds,
-      };
-
-      const result = await profileController.listProfiles(
-        accountSid,
-        { sortDirection, sortBy, limit, offset },
-        { filters },
-      );
-
-      if (isErr(result)) {
-        return next(mapHTTPError(result, { InternalServerError: 500 }));
-      }
-
-      res.json(result.data);
-    } catch (err) {
-      return next(createError(500, err.message));
-    }
-  },
-);
+profilesRouter.get('/', publicEndpoint, async (req: Request, res: Response) => {
+  const { accountSid } = req;
+  const {
+    sortDirection,
+    sortBy,
+    limit,
+    offset,
+    profileFlagIds: encodedProfileFlagIds,
+  } = req.query as Record<string, any>; // TODO: maybe improve this validation
+
+  const profileFlagIds = encodedProfileFlagIds
+    ? decodeURIComponent(encodedProfileFlagIds)
+        .split(',')
+        .map(s => parseInt(s, 10))
+        .filter(v => v && !isNaN(v))
+    : undefined;
+
+  const filters = {
+    profileFlagIds,
+  };
+
+  const result = await profileController.listProfiles(
+    accountSid,
+    { sortDirection, sortBy, limit, offset },
+    { filters },
+  );
+
+  res.json(result);
```

Comment:
Big important note here: This will change the shape of the payloads we send to the client, from `{id: 1, ... }` to `{ data: {id: 1, ... } }`. If we don't change the client code to account for this update we'll break it - unless I'm missing something?

Reply:
The service layer and data access layer methods this calls were changed to return plain objects rather than Result types, so we need to send the top level object in the payload

Also, I would be a little troubled if I had broken the contract of an endpoint so fundamentally and no service tests failed... :-P

Reply:
>Also, I would be a little troubled if I had broken the contract of an endpoint so fundamentally and no service tests failed... :-P

Oh yeah, I totally reviewed the changes in the tests ... :eyes: haha